src/celeritas/optical/MaterialParams.cc

```diff
@@ -120,8 +120,8 @@ MaterialParams::MaterialParams(Input const& inp)
         CELER_VALIDATE(is_monotonic_increasing(make_span(ri.x)),
                        << "refractive index energy grid values are not "
                           "monotonically increasing");
-        CELER_VALIDATE(is_monotonic_increasing(make_span(ri.y)),
-                       << "refractive index values are not monotonically "
+        CELER_VALIDATE(is_monotonic_nondecreasing(make_span(ri.y)),
```

Comment:
Our Cherenkov dn/dx calculator uses an inverse grid (flips the energy and rindex values) to find the energy at a given rindex, relying on both being monotonic increasing. Won't this change mean we could create a grid with constant x values and increasing y values?

Reply:
That was my concern. But maybe it's ok to have coincident points that result in a discontinuity? Would be worth checking in a unit test of the inverse grid I think...

Reply:
One issue I see is our `Interpolator` currently expects the left grid point to be strictly less than the right, and can result in nan values if not.

Reply:
But we can't eve be "inside" a grid cell with zero width can we? I think a test as part of this PR would be great.

Reply:
I don't think we have a special case in the grid calculator for landing exactly on a grid point and skipping interpolation, we only do that if we're on one of the grid endpoints (or out of bounds).

Reply:
This should be fixed with https://github.com/celeritas-project/celeritas/pull/1906.